### PR TITLE
[master]Modular Systems: Document saltenvs and modules

### PR DESCRIPTION
**Pull request by AstraLuma**
### What does this PR do?

Documents how saltenvs and modules interact.

This is a docs-only change.

### CODE REVIEW

**Comment by astronouth7303** [comment]
I feel like it's phrased super awkwardly?

File: doc/topics/development/modules/index.rst [hunk 1]
```diff
@@ -78,6 +75,12 @@ or specific dynamic modules. The ``saltutil.sync_*``
 :py:mod:`runner functions <salt.runners.saltutil>` can be used to sync modules
 to minions and the master, respectively.
 
+Which saltenvs that the minion syncs from is based on the topfiles. (Unless specified manually.)
+
+The minion syncs modules from saltenvs that mention the minion in the saltenv's topfile.
+
+That is, for each saltenv on the file server, if a minion is mentioned in that saltenv's topfile, the minion will load modules from that saltenv.
+
```
Comment (mchugh19) [inline doc/topics/development/modules/index.rst]: How about something a bit more direct, like:

>If saltenv environments are used (through the :ref:`top file <states-top>`, the :conf_minion:`environment` option of the minion configuration file, or as an argument on the command line) modules will be synced from the applied environments.

**Review by Akm0d — APPROVED** [review]

**Comment by astronouth7303** [comment]
I think this is good to go?

**Comment by astronouth7303** [comment]
What's up with this?

**Pull request merged**